From: Kai-Uwe Behrmann <ku.b@gmx.de>
Date: Tue, 26 Mar 2013 11:09:01 +0000 (+0100)
Subject: * [source]: omit profile with error
X-Git-Url: http://www.oyranos.org/scm?p=oyranos.git;a=commitdiff_plain;h=ecbfe88cf77863c7c925f61d0e69d11474059cb8

* [source]: omit profile with error

a proper return value would be more helpful
---

diff --git a/src/API_generated/oyProfile_s_.c b/src/API_generated/oyProfile_s_.c
index 83629fe..338ed08 100644
--- a/src/API_generated/oyProfile_s_.c
+++ b/src/API_generated/oyProfile_s_.c
@@ -8,12 +8,12 @@
  *  Oyranos is an open source Colour Management System
  *
  *  @par Copyright:
- *            2004-2012 (C) Kai-Uwe Behrmann
+ *            2004-2013 (C) Kai-Uwe Behrmann
  *
  *  @author   Kai-Uwe Behrmann <ku.b@gmx.de>
  *  @par License:
  *            new BSD - see: http://www.opensource.org/licenses/bsd-license.php
- *  @date     2012/10/31
+ *  @date     2013/03/26
  */
 
 
@@ -724,6 +724,9 @@ oyProfile_s_ *  oyProfile_FromFile_  ( const char        * name,
     }
   }
 
+  if(error >= 1 && s)
+    oyProfile_Release( (oyProfile_s**)&s );
+
   oyHash_Release_( &entry );
 
   return s;
diff --git a/src/sources/Profile.private_methods_definitions.c b/src/sources/Profile.private_methods_definitions.c
index 7b49ca6..e131893 100644
--- a/src/sources/Profile.private_methods_definitions.c
+++ b/src/sources/Profile.private_methods_definitions.c
@@ -311,6 +311,9 @@ oyProfile_s_ *  oyProfile_FromFile_  ( const char        * name,
     }
   }
 
+  if(error >= 1 && s)
+    oyProfile_Release( (oyProfile_s**)&s );
+
   oyHash_Release_( &entry );
 
   return s;

